pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -314,8 +316,16 @@ void refreshSegmentStates() {
       }
       if (!enabledCandidates.isEmpty()) {
         instanceCandidatesMap.put(segment, enabledCandidates);
+      } else {
+        // Do not count new segment as unavailable
```

Comment:
`count new segment as available` would be easier to understand

Reply:
We don't track available segment but only unavailable ones, so IMO it is more clear to comment on whether it is counted as unavailable